src/AndersonAcceleration.f90

```diff
@@ -278,15 +298,63 @@ SUBROUTINE step_AndersonAccelerationType(solver,x_new)
 !> @param x Initial iterate solve is starting from
 !>
 SUBROUTINE reset_AndersonAccelerationType(solver,x)
+  CHARACTER(LEN=*),PARAMETER :: myName='reset_AndersonAccelerationType'
   CLASS(AndersonAccelerationType),INTENT(INOUT) :: solver
-  REAL(SRK),INTENT(IN) :: x(:)
+  CLASS(VectorType),INTENT(IN) :: x
+
+  TYPE(ParamType) :: vecparams
+  INTEGER(SIK) :: i,j,m
 
-  INTEGER(SIK) :: i,j
+  REQUIRE(x%n == solver%n)
 
-  REQUIRE(SIZE(x) == solver%n)
+  !If this is the first call to set/reset must actually create vectors of needed type
```

Comment:
Vector types has a base capability to create a new vector identical to an old vector: `VectorResemble`

Reply:
also... what blas interface do you need that doesn't exist?

Reply:
Okay, I will look at the resemble routine and see about simplifying the reset/set function. For Trilinos vectors it is dot product (as hard as that is to believe I am looking right at the interface, it's got NativeDistributed, Petsc, and Real...no Trilinos). For NativeDistributed it's the _scal routine. Shouldn't be hard to add.

Reply:
On the subject of BLAS interfaces. The vector interfaces to them have their arguments out of order...for instance _axpy for vectors actually has the arguments in the order xya...a little OCD for sure, but annoying.

Reply:
None of this was fixed yet

Reply:
Latest push adds resemble routines for allocatable vectors and arrays of vectors as well as associated testing. Also changed Anderson over to using them with a short check for vector types that are not currently fully supported in BLAS.